### ISSUE TITLE
With multiple cursors, autocompletion suggests what you're already typing

### ISSUE THREAD

**Issue by atombender** [Microsoft/vscode#17785]
- VSCode Version: 1.8.1

Steps to Reproduce:

* Create document (EDIT @egamma `Plain Text`) with this content:

```
foobar
hey hey
```

* Use Cmd-D to mark both instances of `hey`
* Press `f`
* First autocompletion will be `f`, not `foobar`. This is because it matches itself.
* Press `o`. Autocompletion dropdown disappears. This is due to #4911.
* Press `o`. First autocompletion will `foo`, not `foobar`.
* And so it goes.

In short, it never makes sense for autocompletion to match something you're already in the process of typing.
